src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1711,28 +1711,28 @@ private ArrayTypeSymbol VisitArrayInitializer(BoundArrayCreation node)
                 return arrayType;
             }
 
+            var expressionsNoConversions = ArrayBuilder<BoundExpression>.GetInstance(n);
             var conversions = ArrayBuilder<Conversion>.GetInstance(n);
             var resultTypes = ArrayBuilder<TypeWithState>.GetInstance(n);
             for (int i = 0; i < n; i++)
```

Comment:
Now that we're only using `i` in 1 place, consider making this a a foreach.

Reply:
And there are other loops in this function that could be similarly converted.

---
In reply to: [267007162](https://github.com/dotnet/roslyn/pull/34222#discussion_r267007162) [](ancestors = 267007162)